ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -602,6 +603,62 @@ int mca_pml_ucx_isend(const void *buf, size_t count, ompi_datatype_t *datatype,
     }
 }
 
+static int 
+mca_pml_ucx_bsend(ucp_ep_h ep, const void *buf, size_t count, 
+                  ompi_datatype_t *datatype, int dst, int tag, 
+                  struct ompi_communicator_t* comm)
+{
+    ompi_request_t *req;
+    void *packed_data;
+    size_t packed_length;
+    size_t offset;
+    uint32_t iov_count;
+    struct iovec iov;
+    opal_convertor_t opal_conv;
+
+    OBJ_CONSTRUCT(&opal_conv, opal_convertor_t);
+    opal_convertor_copy_and_prepare_for_recv(ompi_proc_local_proc->super.proc_convertor,
+                                             &datatype->super, count, buf, 0,
+                                             &opal_conv);
+    opal_convertor_get_packed_size(&opal_conv, &packed_length);
+
+    packed_data = mca_pml_base_bsend_request_alloc_buf(packed_length);
+    if (NULL == packed_data) {
+        OBJ_DESTRUCT(&opal_conv);
+        return OMPI_ERR_OUT_OF_RESOURCE;
+    }
+    
+    iov_count    = 1;
+    iov.iov_base = packed_data;
+    iov.iov_len  = packed_length;
+
+    offset = 0;
+    opal_convertor_set_position(&opal_conv, &offset);
+    opal_convertor_pack(&opal_conv, &iov, &iov_count, &packed_length);
+    OBJ_DESTRUCT(&opal_conv);
+
+    req = (ompi_request_t*)ucp_tag_send_nb(ep, packed_data, packed_length,
+                                           ucp_dt_make_contig(1),
+                                           PML_UCX_MAKE_SEND_TAG(tag, comm),
+                                           mca_pml_ucx_bsend_completion);
+    if (OPAL_UNLIKELY(UCS_PTR_IS_ERR(req))) {
+        /* relese temp buffer, print error */
+        mca_pml_base_bsend_request_free(packed_data);
+        PML_UCX_ERROR("ucx bsend failed: %s", ucs_status_string(UCS_PTR_STATUS(req)));
+        return OMPI_ERROR;
+    }
+
+    if (req == NULL) {
+        /* request was completed in place */
+        mca_pml_base_bsend_request_free(packed_data);
+        return OMPI_SUCCESS;
+    }
+
+    ucp_worker_progress(ompi_pml_ucx.ucp_worker);
```

Comment:
i don't think it is needed. But since pml_ucx_send() uses progress after doing ucx send i added it here too.